### PR TITLE
Deployd leader problems

### PR DESCRIPTION
**Pull request by mattmb**
* Fix a bug where the startup stalls and process doesn't exit on abdication
* Add some logging to help catch other leadership bugs

### CODE REVIEW

File: paasta_tools/deployd/leader.py [hunk 1]
```diff
@@ -54,5 +54,14 @@ def reconnection_listener(self):
             time.sleep(5)
             attempts += 1
         self.log.error("Connection did not recover, abdicating!")
+        self.suicide()
+
+    def suicide(self):
```
Comment (Rob-Johnson) [inline paasta_tools/deployd/leader.py]: can we name this something else? ``def terminate`` should be ok (maybe even ``_terminate`` to indicate this isn't really a public api)

File: paasta_tools/deployd/master.py [hunk 2]
```diff
@@ -286,9 +288,21 @@ def start_watchers(self):
...
+            if all([watcher.is_ready for watcher in self.watcher_threads]):
+                return
+            self.log.info("Sleeping and waiting for watchers to all start")
+            self.log.info([
+                {
+                    'name': watcher.__class__.__name__,
+                    'ready': watcher.is_ready,
+                } for watcher in self.watcher_threads
+            ])
+            time.sleep(5)
```
Comment (Rob-Johnson) [inline paasta_tools/deployd/master.py]: this seems a long time? I'm kind of expecting these workers to start quickly - what could lead to them taking so long?

Comment (mattmb) [reply paasta_tools/deployd/master.py]: Yeah, I wasn't quite sure what was reasonable. I think we have to get a watch on all the ZK nodes but I guess that is quick. Any reason not to be conservative though?

File: paasta_tools/deployd/master.py [hunk 3]
```diff
@@ -286,9 +288,21 @@ def start_watchers(self):
...
-            self.log.debug("Sleeping and waiting for watchers to all start")
+        attempts = 0
+        while attempts < 120:
+            if all([watcher.is_ready for watcher in self.watcher_threads]):
+                return
+            self.log.info("Sleeping and waiting for watchers to all start")
+            self.log.info([
+                {
+                    'name': watcher.__class__.__name__,
+                    'ready': watcher.is_ready,
```
Comment (keymone) [inline paasta_tools/deployd/master.py]: maybe lets only log watchers that aren't ready?

**Pull request merged**